File: py/cli/src/tuish_cli/commands/analytics.py
from __future__ import annotations

import typer
from rich.console import Console

from tuish_cli.config import load_config, require_api_key
from tuish_cli.output import emit_not_implemented

console = Console()


def analytics(
    ctx: typer.Context,
    period: str | None = typer.Option(None, "--period", help="Time window (e.g. 7d, 30d)"),
) -> None:
    config = load_config(ctx.obj.config_path)
    require_api_key(ctx.obj, config)

    if ctx.obj.json:
        extra = {"period": period} if period else None
        emit_not_implemented(
            "Analytics",
            "Analytics will be added once the API is available.",
            extra=extra,
        )
        return

    console.print("[bold]Analytics[/bold]")
    console.print("[yellow]Not implemented yet[/yellow]")
    if period:
        console.print(f"[dim]Period: {period}[/dim]")
    else:
        console.print("[dim]Analytics will be added once the API is available.[/dim]")

File: rs/cli/src/commands/analytics.rs
use super::{load_config, print_placeholder, require_api_key, CliResult, Context};
use serde_json;

pub fn run(ctx: &Context, period: Option<String>) -> CliResult {
	let config = load_config(&ctx.config_path)?;
	require_api_key(&config)?;

	if let Some(period) = period {
		if ctx.json {
			let payload = serde_json::json!({
				"status": "not_implemented",
				"title": "Analytics",
				"message": "Analytics will be added once the API is available.",
				"period": period,
			});
			return super::output_json(&payload);
		}
		print_placeholder(ctx, "Analytics", &format!("Period: {period}"))?;
		return Ok(());
	}
	print_placeholder(ctx, "Analytics", "Analytics will be added once the API is available.")?;
	Ok(())
}
